### DISCUSSION TITLE
user-defined placeholders

### DISCUSSION THREAD

**Discussion by Blueeyestar** [Zrips/CMI#5380] [Enhancement]
Good evening,
we wanted to know whether there are already user-defined placeholders at CMI.
The reason is as follows.
We are running a roulette event on the server. Only users who have registered are allowed to participate.
Each player's account balance is set to 200 thalers.
Now we want to create a board with the _AdvancedGUI | plugin Interactive ItemFrame_.
Users should see what amount user X still has in the tournament. That's why we need a money placeholder where you can assign users individually.

as an example **%cmi_**USER**_balance%**   <<---With user you have to enter the name of the player manually in the placeholder

Because then you would have to supplement the placeholder
**% cmi_Blueeyestar_balance%**

With the placeholder, other users would see exactly how much credit the player still has.

Is that possible or is it too complicated?

**Comment by CloudeLecaw** [comment]
I agree this would be a nice extension to the current placeholders that show just the used players stats. Also this would make it feel better in its handling than using a papi extension for this.
So please add poosibillity to add placeholders like
`%cmi_user_[playerName]_balance% `  option
example from abouve would be
`%cmi_user_Blueeyestar_balance%`
This would be specially usefull if you want to show stats of others to others

Would be nice to see something like that for many more place holders
we have so many nice cmi user placeholders where this might be usefull

```
%cmi_user_[playerName]_charges_left%
%cmi_user_[playerName]_charges_max%
%cmi_user_[playerName]_charges_time%
%cmi_user_[playerName]_charges_cooldown%
%cmi_user_[playerName]_display_name%
%cmi_user_[playerName]_nickname%
%cmi_user_[playerName]_name%
%cmi_user_[playerName]_deathloc%
%cmi_user_[playerName]_cuffed%
%cmi_user_[playerName]_muted%
%cmi_user_[playerName]_inpvp%
%cmi_user_[playerName]_god%
%cmi_user_[playerName]_afk%
%cmi_user_[playerName]_afk_symbol%
%cmi_user_[playerName]_afk_msg%
%cmi_user_[playerName]_afk_for%
%cmi_user_[playerName]_afk_in%
%cmi_user_[playerName]_joinedcounter%
%cmi_user_[playerName]_banned%
%cmi_user_[playerName]_maxhomes%
%cmi_user_[playerName]_ping%
%cmi_user_[playerName]_gamemode%
%cmi_user_[playerName]_op%
%cmi_user_[playerName]_canfly%
%cmi_user_[playerName]_flying%
%cmi_user_[playerName]_homeamount%
%cmi_user_[playerName]_vanished_symbol%
%cmi_user_[playerName]_balance_formated%
%cmi_user_[playerName]_balance_formatted%
%cmi_user_[playerName]_balance%
%cmi_user_[playerName]_prefix%
%cmi_user_[playerName]_suffix%
%cmi_user_[playerName]_nameplate_prefix%
%cmi_user_[playerName]_nameplate_suffix%
%cmi_user_[playerName]_group%
%cmi_user_[playerName]_tfly%
%cmi_user_[playerName]_tfly_formated%
%cmi_user_[playerName]_tfly_formatted%
%cmi_user_[playerName]_flightcharge%
%cmi_user_[playerName]_tgod%
%cmi_user_[playerName]_tgod_formated%
%cmi_user_[playerName]_tgod_formatted%
%cmi_user_[playerName]_votecount%
%cmi_user_[playerName]_rank%
%cmi_user_[playerName]_nextranks%
%cmi_user_[playerName]_nextrankpercent%
%cmi_user_[playerName]_nextvalidranks%
%cmi_user_[playerName]_canrankup%
%cmi_user_[playerName]_country%
%cmi_user_[playerName]_country_code%
%cmi_user_[playerName]_city%
%cmi_user_[playerName]_name_colorcode%
%cmi_user_[playerName]_glow_code%
%cmi_user_[playerName]_glow_name%
%cmi_user_[playerName]_jailname%
%cmi_user_[playerName]_jailcell%
%cmi_user_[playerName]_jailtime%
%cmi_user_[playerName]_jailreason%
%cmi_user_[playerName]_bungeeserver%
%cmi_user_[playerName]_playtime_formatted%
%cmi_user_[playerName]_playtime_days%
%cmi_user_[playerName]_playtime_dayst%
%cmi_user_[playerName]_playtime_hours%
%cmi_user_[playerName]_playtime_hoursf%
%cmi_user_[playerName]_playtime_hourst%
%cmi_user_[playerName]_playtime_minutes%
%cmi_user_[playerName]_playtime_minutest%
%cmi_user_[playerName]_playtime_seconds%
%cmi_user_[playerName]_playtime_secondst%
%cmi_user_[playerName]_itemcount_[itemIdName(:data)]%
%cmi_user_[playerName]_maxperm_[corePerm]_[defaultValue]%
```

**Comment by Zrips** [comment]
What about PAPI %parseother_{UserName}_{placeholder}% ?
